spring-modulith-events/spring-modulith-events-mongodb/src/test/java/org/springframework/modulith/events/mongodb/MongoDbEventPublicationRepositoryTest.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.modulith.events.mongodb;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.within;
+
+import java.io.IOException;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
+import lombok.Getter;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.modulith.events.CompletableEventPublication;
+import org.springframework.modulith.events.EventPublication;
+import org.springframework.modulith.events.PublicationTargetIdentifier;
+import org.springframework.modulith.testapp.TestApplication;
+import org.springframework.test.context.ContextConfiguration;
+
+import com.mongodb.client.MongoClients;
+
+import de.flapdoodle.embed.mongo.MongodExecutable;
+import de.flapdoodle.embed.mongo.MongodStarter;
+import de.flapdoodle.embed.mongo.config.ImmutableMongodConfig;
+import de.flapdoodle.embed.mongo.config.MongodConfig;
+import de.flapdoodle.embed.mongo.config.Net;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import de.flapdoodle.embed.process.runtime.Network;
+import lombok.EqualsAndHashCode;
+
+/**
+ * @author Björn Kieling
+ * @author Dmitry Belyaev
+ */
+@DataMongoTest
+@ContextConfiguration(classes = TestApplication.class)
+class MongoDbEventPublicationRepositoryTest {
+
+	private static final PublicationTargetIdentifier TARGET_IDENTIFIER = PublicationTargetIdentifier.of("listener");
+	private static final String CONNECTION_STRING = "mongodb://%s:%d";
+
+	private static String ip;
+	private static int port;
+	private static MongodExecutable mongodExecutable;
+
+	private MongoTemplate mongoTemplate;
+
+	private MongoDbEventPublicationRepository repository;
+
+	@BeforeAll
```

Comment:
We should be able to use the Flapdoodle Spring Boot autoconfiguration in version [1.1.0-spring27x](https://search.maven.org/artifact/de.flapdoodle.embed/de.flapdoodle.embed.mongo.spring/1.1.0-spring27x/jar) to avoid the need for all that infrastructure setup. Note, that the setup shown in the [how-to guide](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo.spring/blob/spring-2.7.x/HowTo.md) can even be simplified, as Boot has removed the autoconfiguration the sample code explicitly excludes.

Reply:
Unfortunately this solution doesn't work with Spring Boot 3.x